### PR TITLE
y axis responsive labels

### PR DESCRIPTION
**Pull request by pranjaljately**

### CODE REVIEW

File: src/components/chart.svelte [hunk 1]
```diff
@@ -28,9 +28,9 @@
         ctx.beginPath()
         ctx.lineWidth = LINE_WIDTH
         ctx.strokeStyle = BORDER_COLOR
-        ctx.fillStyle = BORDER_BG_COLOR
-        ctx.rect(0, 0, newWidth, newHeight)
-        ctx.fillRect(0, 0, newWidth, newHeight)
+        ctx.fillStyle = CHART_BG_COLOR
+        ctx.rect(30, 0, newWidth, newHeight)
```
Comment (zubhav) [inline src/components/chart.svelte]: As discussed, lets create a Y_AXIS_OFFSET const to store this 30

File: src/components/chart.svelte [hunk 2]
```diff
@@ -39,22 +39,24 @@
         let minYValue = Math.min(...allValues)
         let maxYValue = Math.max(...allValues)
 
+        drawYLabels(minYValue, maxYValue, 10)
```
Comment (zubhav) [inline src/components/chart.svelte]: Shall we call this within redrawGraph function instead?

File: src/components/chart.svelte [hunk 3]
```diff
@@ -91,6 +93,38 @@
         }
     }
 
+    function calculateMaxYValue(maxY, steps) {
+        if (maxY < 2) return Math.ceil(maxY)
+        const roundedToTen = Math.ceil(maxY / 10) * 10
+        return Math.ceil(roundedToTen / steps) * steps
+    }
+
+    function calculateMinYValue(minY, steps) {
```
Comment (zubhav) [inline src/components/chart.svelte]: Merge these two functions into one and pass Math.floor and Math.ceil as param called `roundFunc`? I think that is valid

File: src/components/chart.svelte (hunk 2 above)
Comment (zubhav) [reply src/components/chart.svelte]: Chart component should accept `ticks` and `let Y_AXIS_TICKS = ticks || 10` should be globalised.

File: src/components/chart.svelte [hunk 4]
```diff
@@ -91,6 +93,38 @@
...
+        const roundedToTen = Math.floor(minY / 10) * 10
+        return Math.floor(roundedToTen / steps) * steps
+    }
+
+    function drawYLabels(min, max, steps) {
+        let maxYValue = calculateMaxYValue(max, steps)
+        let minYValue = calculateMinYValue(min, steps)
+        let range = maxYValue - minYValue
+        let yFrequency = range / (steps - 1)
+        let ySpacing = (chartHeight - 7) / (steps - 1)
```
Comment (zubhav) [inline src/components/chart.svelte]: const Y_AXIS_SPACING = 7 at the top of the file with other constants 
const segments = steps - 1 at the top of this function?

^ at the top

File: src/components/chart.svelte (hunk 2 above)
Comment (pranjaljately) [reply src/components/chart.svelte]: Sure. It needs the min and max Y values to be supplied or could create a function that returns them (as they are required in drawGraph as well.
`
function getMinandMaxValues(values) {
        const allValues = valueSet.flat()
        let min = Math.min(...allValues)
        let max = Math.max(...allValues)
        return {min, max}
}
`

**Pull request merged**